test/fixtures/dialects/ansi/arithmetic_a.yml

```diff
@@ -20,7 +20,9 @@ file:
               - binary_operator: '*'
               - literal: '3'
               end_bracket: )
-          - comparison_operator: '>='
+          - comparison_operator:
```

Comment:
I notice the grouping segment and the items inside it have the same type. Is this potentially tricky for rule writers?

Reply:
@barrywhart 
I would definitely want a lone `>` to have the same type as the `>=` grouping.
I guess the easy solution in rules is just say find a segment of type `comparison_operator` whose parent is not also a `comparison_operator` and infer it's kind via the name.
(it's a shame we don't store a parent attribute on the segments (like a linked list) rather than the context parent stack method because then you could just say `if segment.is_type("comparison_operator") and not segment.parent.is_type("comparison_operator")`, as well as doing any other required traversing without requiring context)

I think its a better solution that the previous situation where we had previously for T-SQL:
```
- comparison_operator: '>'
- whitespace: ' '
- comparison_operator: '='
```
since we had no obvious grouping or programatic way to refer to this as `greater_than_equal_to`.

Is there an easy way to overwrite the type of the segments within the grouping, but not the lone `>`? 

If not potentially something like this would work:
```python
RawGreaterThanSegment=StringParser(
    ">", SymbolSegment, name="raw_greater_than", type="raw_comparison_operator"
),
RawEqualsSegment=StringParser(
    "=", SymbolSegment, name="raw_equals", type="raw_comparison_operator"
)
...
@ansi_dialect.segment()
class EqualsSegment(BaseSegment):
    """equals operator."""

    type = "comparison_operator"
    name = "equals"
    match_grammar = Ref("RawEqualsSegment")

@ansi_dialect.segment()
class GreaterThanSegment(BaseSegment):
    """Greater than operator."""

    type = "comparison_operator"
    name = "greater_than"
    match_grammar = Ref("RawGreaterThanSegment")

@ansi_dialect.segment()
class GreaterThanOrEqualToSegment(BaseSegment):
    """Greater than or equal to operator."""

    type = "comparison_operator"
    name = "greater_than_equal_to"
    match_grammar = Sequence(
        Ref("GreaterThanSegment"), Ref("EqualsSegment"), allow_gaps=False
    )
```
That way `segment.type = "comparison_operator"` only matches the grouped segments and also makes single character comparison operators consistent AST-wise:
```
# Equals
- comparison_operator:
    - raw_comparison_operator: '='
# Greater Than
- comparison_operator:
    - raw_comparison_operator: '>'
# Greater Than or Equal To
- comparison_operator:
    - raw_comparison_operator: '>'
    - raw_comparison_operator: '='
```
Main question there is if you think that's too verbose vs just including a condition in the rules?

Reply:
I don't have a strong opinion either way. This feels like a classic tradeoff -- make X more complex to make Y less complex. On the one hand, SQLFluff has many dialects, but on the other hand, it also has lots of rules. I'm also unsure how many rules will be specifically dealing with comparison operators.

I'm okay leaving it "as is" for now. We can adjust later if needed as we develop and maintain rules.

Your proposal for how to do it using the grammar looks like it would work well.

Reply:
yeah I think that's fair enough, the raw suggestion shouldn't be too hard to implement but lets hold off until we actually have a need 👍

Reply:
Going to look at this as I think it's worth keeping it simple on the rules side